airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py

```diff
@@ -0,0 +1,70 @@
+import sys
+from typing import Any, Callable, Optional
+
+import dagger
+from asyncclick import Context, get_current_context
+from dagger.api.gen import Client, Container
+from pydantic import BaseModel, Field, PrivateAttr
+
+from ..singleton import Singleton
+
+
+class ClickPipelineContext(BaseModel, Singleton):
+    dockerd_service: Optional[Container] = Field(default=None)
+    _dagger_client: Optional[Client] = PrivateAttr(default=None)
+    _click_context: Callable[[], Context] = PrivateAttr(default_factory=lambda: get_current_context)
+
+    @property
+    def params(self):
+        ctx = self._click_context()
+        click_obj = ctx.obj
+        click_params = ctx.params
+        command_name = ctx.command.name
+
+        # Error if click_obj and click_params have the same key
+        all_click_params_keys = [p.name for p in ctx.command.params]
+        intersection = set(click_obj.keys()) & set(all_click_params_keys)
+        if intersection:
+            raise ValueError(f"Your command '{command_name}' has defined options/arguments with the same key as its parent: {intersection}")
+
+        return {**click_obj, **click_params}
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def __init__(self, **data: dict[str, Any]):
+        """
+        Initialize the ClickPipelineContext instance.
+
+        This method checks the _initialized flag for the ClickPipelineContext class in the Singleton base class.
+        If the flag is False, the initialization logic is executed and the flag is set to True.
+        If the flag is True, the initialization logic is skipped.
+
+        This ensures that the initialization logic is only executed once, even if the ClickPipelineContext instance is retrieved multiple times.
+        This can be useful if the initialization logic is expensive (e.g., it involves network requests or database queries).
+        """
+        if not Singleton._initialized[ClickPipelineContext]:
+            super().__init__(**data)
+            Singleton._initialized[ClickPipelineContext] = True
+
+    import asyncio
```

Comment:
Can we avoid the inline import? 
I would also prefer to use `anyio` primitives instead of `asyncio`

Reply:
Sure thing. Whats the reason for prefering anyio?

Reply:
`anyio` is the async framework wrapper that dagger suggested that we use. It can run on top of `asyncio` or `trio`. And  in tests, `anyio` and `asyncio` fixture do not play well together.